### PR TITLE
Support multi-reigon test

### PR DESCRIPTION
**Pull request by ekexium**
Previously bugs with multiple regions are hard to find in tests (e.g. #251), we should improve our test to cover such situation.

Changes:
- Support integration tests with multiple regions
- Improve the `write_million` tests since they don't have to write much data to make region split. As a result tests should be faster

How it works:
- Use specific config for TiKV and PD to enforce region splitting
- Enforce region splitting in integration tests when `MULTI_REGION=1` is set

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -48,13 +48,15 @@ script:
...
   - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_RUST_VERSION == "nightly" ]]; then docker ps; fi
   - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_RUST_VERSION == "nightly" ]]; then docker logs pd; fi
   - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_RUST_VERSION == "nightly" ]]; then docker logs kv; fi
   - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_RUST_VERSION == "nightly" ]]; then sleep 60; fi
-  - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_RUST_VERSION == "nightly" ]]; then PD_ADDRS="127.0.0.1:2379" cargo test --all --features integration-tests -- --nocapture; fi
+  - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_RUST_VERSION == "nightly" ]]; then PD_ADDRS="127.0.0.1:2379" MULTI_REGION=1 cargo test txn_ --all --features integration-tests -- --nocapture; fi
+  - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_RUST_VERSION == "nightly" ]]; then PD_ADDRS="127.0.0.1:2379" MULTI_REGION=1 cargo test raw_ --all --features integration-tests -- --nocapture; fi
+  - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_RUST_VERSION == "nightly" ]]; then PD_ADDRS="127.0.0.1:2379" MULTI_REGION=1 cargo test misc_ --all --features integration-tests -- --nocapture; fi
```
Comment (nrc) [inline .travis.yml:59]: can we use make on Travis?

File: tests/common/mod.rs [hunk 2]
```diff
@@ -14,7 +21,59 @@ pub async fn clear_tikv() {
...
+            .map(|x| x.to_be_bytes().to_vec());
+
+        ensure_region_splitted(keys_1.chain(keys_2), 100).await?;
+    }
+
+    clear_tikv().await;
+    Ok(())
+}
+
+async fn ensure_region_splitted(
```
Comment (nrc) [inline tests/common/mod.rs:46]:
```suggestion
async fn ensure_region_split(
```

File: tests/common/mod.rs [hunk 3]
```diff
@@ -14,7 +21,59 @@ pub async fn clear_tikv() {
...
+    keys: impl IntoIterator<Item = impl Into<Key>>,
+    region_count: u32,
+) -> Result<()> {
+    if ctl::get_region_count().await? as u32 >= region_count {
+        return Ok(());
+    }
+
+    // 1. write plenty transactional keys
+    // 2. wait until regions splitted
```
Comment (nrc) [inline tests/common/mod.rs:55]:
```suggestion
    // 2. wait until regions split
```

File: tests/common/mod.rs [hunk 4]
```diff
@@ -14,7 +21,59 @@ pub async fn clear_tikv() {
...
+    for key in keys.into_iter() {
+        txn.put(key.into(), vec![0, 0, 0, 0]).await?;
+    }
+    txn.commit().await?;
+    let mut txn = client.begin_optimistic().await?;
+    let _ = txn.scan(vec![].., 2048).await?;
+    txn.commit().await?;
+
+    info!("splitting regions...");
+    loop {
```
Comment (nrc) [inline tests/common/mod.rs:68]: Should we have a safety check exit in case PD never splits the regions?

File: tests/common/mod.rs [hunk 5]
```diff
@@ -23,3 +82,10 @@ pub fn pd_addrs() -> Vec<String> {
         .map(From::from)
         .collect()
 }
+
+fn enable_multi_region() -> bool {
+    match env::var(ENV_ENABLE_MULIT_REGION) {
+        Ok(s) => s == "1" || s.to_lowercase() == "true",
```
Comment (nrc) [inline tests/common/mod.rs:88]: I think it is usual practice to enable a feature if the env var is set to anything

Comment (ekexium) [reply tests/common/mod.rs:88]: That's my first thought too. Then I thought `MULTI_REGION=0` would be weird. I don't know a lot about the convention here.

File: tests/common/mod.rs (hunk 4 above)
Comment (ekexium) [reply tests/common/mod.rs:68]: Good idea.

File: tests/common/mod.rs (hunk 5 above)
Comment (nrc) [reply tests/common/mod.rs:99]: I think you must unset an env var not set it to 0 if you want it to be falsey

Comment (ekexium) [reply tests/common/mod.rs]: OK

**Review by nrc — APPROVED** [review]
lgtm, thanks for the changes

**Pull request merged**